gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -455,6 +471,33 @@ public synchronized void setActive(boolean active) {
     }
   }
 
+  /**
+   * Used by the DagManager to launch a new execution for a flow action event loaded from the DagActionStore upon
+   * setting this instance of the DagManager to active. Because it may be a completely new DAG not contained in the
+   * dagStore, we compile the flow to generate the dag before calling addDag(), handling any errors that may result in
+   * the process.
+   */
+  public void handleLaunchFlowEvent(DagActionStore.DagAction action) {
+    FlowId flowId = action.getFlowId();
+    FlowSpec spec;
+    try {
+      URI flowUri = FlowSpec.Utils.createFlowSpecUri(flowId);
+      spec = (FlowSpec) flowCatalog.getSpecs(flowUri);
+      Optional<Dag<JobExecutionPlan>> optionalJobExecutionPlanDag = orchestrator.handleChecksBeforeExecution(spec);
```

Comment:
I'm thinking since we need to pass the orchestrator to the dagmanager here which seems like an antipattern, it would be easier to have the orchestrator handle the dagmanager launch event on startup. What would the LOE be in that case?

Reply:
Moving this function to the `Orchestrator`, does not actually change the code pattern here since there's only a bit of wrapper code (mainly to get the spec corresponding to the `dagAction`) around the `Orchestrator` helper funcs that it relies on. It's more clear to keep it here since the use case is `DagManager` specific